R/collapse.R

```diff
@@ -23,10 +23,14 @@ fct_collapse <- function(.f, ..., group_other = FALSE) {
     f <- check_factor(.f)
     levels <- levels(f)
     new[["Other"]] <- levels[!levels %in% levs]
-    levs <- levels
+    levs <- as.list(unlist(new, use.names = FALSE))
```

Comment:
Can you explain what this is doing? Unlisting and relisting an object seems a bit weird.

Reply:
It was just to flatten `new`, and I actually copied it from line 21 in the code (so you'll have to [ask yourself](https://github.com/tidyverse/forcats/blob/4866ef4295d99ed67024ac32ac6d742a407978df/R/collapse.R) :joy:). I did this for consistency but perhaps I should have thought about whether it was the best way.

As it happens, I believe there is no need for `new` to be `as.list`ed, and can just be flattened to a vector. Then, on line 26, `levs` can just be combined with itself and the strings which are going to be collapsed to `other` like so:

```suggestion
    levs <- c(levs, new[["Other"]])
```
Happy to make this change via pull request if it sounds reasonable.

Reply:
Yes, that looks better to be